File: TP/SPRING-MVC/formation-spring-mvc-correction/src/main/java/fr/formation/model/Produit.java
package fr.formation.model;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Positive;

public class Produit {
	private int id;
	
	@NotBlank(message = "Le nom doit être saisi")
	private String nom;
	
	@Positive(message = "Le prix doit être positif")
	private float prix;

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getNom() {
		return nom;
	}

	public void setNom(String nom) {
		this.nom = nom;
	}

	public float getPrix() {
		return prix;
	}

	public void setPrix(float prix) {
		this.prix = prix;
	}
}

File: TP/ANGULAR/formation-angular/src/app/produit.ts
export class Produit {
  public get nom(): string {
    return this._nom;
  }
  
  public set nom(value: string) {
    this._nom = value;
  }
  
  public get prix(): number {
    return this._prix;
  }
  
  public set prix(value: number) {
    this._prix = value;
  }

  constructor(private _nom: string, private _prix: number) { }
}
